pkg/prometheus/promcfg.go

```diff
@@ -2179,6 +2179,10 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		cfg = append(cfg, yaml.MapItem{Key: "metrics_path", Value: *sc.Spec.MetricsPath})
 	}
 
+	if sc.Spec.Params != nil {
+		cfg = append(cfg, yaml.MapItem{Key: "params", Value: sc.Spec.Params})
```

Comment:
Should we sort the params argument by the key to ensure we will always render the config on the same structure?
I did the same [here](https://github.com/prometheus-operator/prometheus-operator/pull/5709#discussion_r1252733778).

Reply:
IIUC by default Go prints the map with sorted keys but while iterating over a map, it follows the order of the keys appearing as it is (an example here https://go.dev/play/p/YeF128GxX2e). Since we are just passing the value here do we need to sort here?

Reply:
@slashpai you should be able to leverage `stringMapToMapSlice()` after https://github.com/prometheus-operator/prometheus-operator/pull/5770 merges.